From: Modestas Vainius <modax@debian.org>
Subject: Fix FBTFS on arm*
Last-Update: 2011-04-24
Forwarded: no

generator gets the signature of QEasingCurve::setCustomType() wrong. What is
more, qreal is involved in the signature. qreal is float on arm* so exclude
setCustomType(float) methods as well.

--- a/generator/typesystem_core-common.xml
+++ b/generator/typesystem_core-common.xml
@@ -921,6 +921,7 @@
   <value-type name="QEasingCurve">
     <modify-function signature="customType()const" remove="all" />
     <modify-function signature="setCustomType(double)" remove="all" />
+    <modify-function signature="setCustomType(float)" remove="all" />
   </value-type>
 
   <object-type name="QAbstractAnimation" />
--- a/generator/typesystem_core.xml
+++ b/generator/typesystem_core.xml
@@ -1388,14 +1388,12 @@
   </value-type>
 
   <value-type name="QEasingCurve">
-    <modify-function signature="customType()const" remove="all"/>
-    <modify-function signature="setCustomType(double)" remove="all"/>
-  
     <modify-function signature="QEasingCurve(QEasingCurve)" remove="all"/>
     <modify-function signature="operator=(QEasingCurve)" remove="all"/>
     <modify-function signature="operator==(const QEasingCurve &amp;)const" remove="all"/>
     <modify-function signature="operator!=(const QEasingCurve &amp;)const" remove="all"/>
     <modify-function signature="setCustomType(double)" remove="all"/>
+    <modify-function signature="setCustomType(float)" remove="all"/>
     <modify-function signature="customType()const" remove="all"/>
   </value-type>
 
--- a/generator/typesystem_core-qtscript.xml
+++ b/generator/typesystem_core-qtscript.xml
@@ -1447,6 +1447,7 @@
     <modify-function signature="operator==(const QEasingCurve &amp;)const" remove="all"/>
     <modify-function signature="operator!=(const QEasingCurve &amp;)const" remove="all"/>
     <modify-function signature="setCustomType(double)" remove="all"/>
+    <modify-function signature="setCustomType(float)" remove="all"/>
     <modify-function signature="customType()const" remove="all"/>
   </value-type>
 
